### PR TITLE
Multiple GTM-container support

### PR DESCRIPTION
**Pull request by postalservice14**
When you have multiple container ID's, you may need different query parameters for each.  This allows you to do so in a backwards-compatible way.

Another way I thought about doing it was adding another config key called `containers` so as not to muddy up or confuse what the `id` key is.

### CODE REVIEW

**Comment by Shinigami92** [comment]
Thanks for the PR :+1: 
I hope I will find today (later this evening) to review it

File: src/plugin.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export interface VueGtmTrackEventParams {
  * Plugin main class
  */
 export default class VueGtmPlugin {
-  constructor(public readonly id: string | string[]) {}
+  constructor(public readonly id: string | string[] | VueGtmContainer[]) {}
```
Comment (Shinigami92) [inline src/plugin.ts:20]: I think we can simplify this constructor to just take 
`constructor(public readonly idContainers: VueGtmContainer[])`

And then we should add a new function that converts a single `id` (`string`), an array of `ids` (`string[]`) or an array of containers to an array of containers.

This function can then also be used before passing to the `loadScript` function
So we don't call it on three different branches but one.

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
I'm also thinking about to only allow `VueGtmContainer | VueGtmContainer[]` as input and then remove the `queryParams` from `VueGtmUseOptions`

And thinking this idea further... does it make sense in any way to just pass the whole configuration multiple times?
Does it anything different than just using multiple VueGtm instances at the same time? ... Yeah :thinking: you can `enable`/`disable` it for every config at once ...

Do I understand it correctly that the multiple container configuration just send the data to different GTMs with the same call? What is the use-case?
Do I want to send sometimes something to the first container and the next time to another container?
So we I need a filter-logic for e.g. `trackView`/`trackEvent`?

---

Aside my headaches :laughing: -> really cool that you already extended the unit tests :+1:

**Comment by Shinigami92** [comment]
Oh and it's totally okay to introduce breaking changes
I already think about to add breaking changes myself for the next version

**Comment by Shinigami92** [comment]
I also updated the README.md in master, so you need to merge the master into your branch
The example now uses valid JS code so the prettier formatter works correctly
So only the simple configuration should be shown and the different other possibilities should be added as a comment

**Comment by postalservice14** [comment]
Thanks for the review!  In our use case, we will programmatically add the containers we want at runtime.  And yes, for us, we will use the same dataLayer so the events get posted to all the containers.  Others might have different use cases, but I think we could think about those as they come up.  So for me, I don't need any filter logic.  Again, maybe a requested feature from some in the future 🤷‍♂️ .

If you're open to a new BC-breaking version, then I agree about making the constructor just take the object or array of objects.  I'll make those changes.  Thanks!

**Comment by Shinigami92** [comment]
Okay, I think I will more this for now
Maybe we will just use only-container in later version

Do you need also a Vue-2 release? Or is it okay if it's only a Vue-3 feature for now?
I would need to update the vue-2 branch for that

**Review by Shinigami92 — APPROVED** [review]

**Comment by postalservice14** [comment]
> Okay, I think I will more this for now
> Maybe we will just use only-container in later version
> 
> Do you need also a Vue-2 release? Or is it okay if it's only a Vue-3 feature for now?
> I would need to update the vue-2 branch for that

Thanks!  Yes, we do need Vue2 release.  Here is my Vue2 PR: https://github.com/mib200/vue-gtm/pull/115

**Comment by Shinigami92** [comment]
Mh that's strange  
I try to `import Vue from 'vue'` in jest test, but `Vue` is undefined :eyes:  

Maybe I just ignore the tests for now?

**Comment by Shinigami92** [comment]
@postalservice14 Should be released as `"vue-gtm": "3.3.0-vue2"` (Use fixed version)

Please report me if everything works :slightly_smiling_face:

**Comment by Shinigami92** [comment]
@postalservice14 Already released a `"vue-gtm": "3.3.1-vue2"`
Because the `window.dataLayer` typedef wasn't correctly exported
Now TypeScript can handle it and build again

**Comment by postalservice14** [comment]
Works great!  👍   Thanks @Shinigami92 !

**Pull request merged**